crane/container.go

```diff
@@ -122,7 +122,9 @@ func (c *container) Dependencies() *Dependencies {
 	dependencies := &Dependencies{}
 	for _, link := range c.RunParams.Link() {
 		linkParts = strings.Split(link, ":")
-		dependencies.All = append(dependencies.All, linkParts[0])
+		if !dependencies.includesAsKind(linkParts[0], "link") {
```

Comment:
VolumesFrom is already supported, just below. Have a look at it, my point was just that if we de-duplicate `.All`, we might as well deduplicate `.Link` (like we do for `.All`/`.VolumesFrom`) by guarding them within the same `if` block.